idaes/models_extra/power_generation/unit_models/heater_1D.py

```diff
@@ -0,0 +1,564 @@
+##############################################################################
+# Institute for the Design of Advanced Energy Systems Process Systems
+# Engineering Framework (IDAES PSE Framework) Copyright (c) 2018-2019, by the
+# software owners: The Regents of the University of California, through
+# Lawrence Berkeley National Laboratory,  National Technology & Engineering
+# Solutions of Sandia, LLC, Carnegie Mellon University, West Virginia
+# University Research Corporation, et al. All rights reserved.
+#
+# Please see the files COPYRIGHT.txt and LICENSE.txt for full copyright and
+# license information, respectively. Both files are also available online
+# at the URL "https://github.com/IDAES/idaes-pse".
+##############################################################################
+"""
+1-D Electric Trim Heater Model With Wall Temperatures
+
+Discretization based on tube rows
+"""
+# Import Pyomo libraries
+from pyomo.environ import (
+    assert_optimal_termination,
+    Var,
+    value,
+    units as pyunits,
+)
+from pyomo.common.config import ConfigBlock, ConfigValue, In
+from pyomo.util.calc_var_value import calculate_variable_from_constraint
+
+# Import IDAES cores
+from idaes.core import (
+    ControlVolume1DBlock,
+    declare_process_block_class,
+    MaterialBalanceType,
+    EnergyBalanceType,
+    MomentumBalanceType,
+    FlowDirection,
+    UnitModelBlockData,
+    useDefault,
+)
+import idaes.core.util.scaling as iscale
+from idaes.core.solvers import get_solver
+from idaes.core.util.config import is_physical_parameter_block
+from idaes.core.util.misc import add_object_reference
+import idaes.logger as idaeslog
+from idaes.core.util.tables import create_stream_table_dataframe
+from idaes.core.util.model_statistics import degrees_of_freedom
+from idaes.models_extra.power_generation.unit_models.heat_exchanger_common import (
```

Comment:
You should not be importing underscore functions from a helper module. If these are intended to be imported like this, then this implies they are not private functions.

Reply:
The main rationale for importing underscore functions is to avoid duplicated code without making a contract with the end user to maintain it in its current form indefinitely. If they're not private functions, that implies there's also a layer of testing, etc. that should go into maintaining them.

Reply:
Whilst I do understand what you are trying to do, it still runs against Python coding practices. Also, given this is in `models_extra`, there is implicitly only a minimal contract for maintenance (although most users would not realize this).

One possible alternative would be to set things up so that the model inherits these methods, e.g. either throguh a base class, multiple inheritance or a mix-in class (don't ask me for details on the last one, I just know such things exist).